contracts/margin/external/BucketLender/BucketLenderProxy.sol

```diff
@@ -0,0 +1,260 @@
+/*
+
+    Copyright 2018 dYdX Trading Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+*/
+
+pragma solidity 0.4.24;
+pragma experimental "v0.5.0";
+
+import { WETH9 } from "canonical-weth/contracts/WETH9.sol";
+import { BucketLender } from "./BucketLender.sol";
+import { AdvancedTokenInteract } from "../../../lib/AdvancedTokenInteract.sol";
+import { MathHelpers } from "../../../lib/MathHelpers.sol";
+import { TokenInteract } from "../../../lib/TokenInteract.sol";
+
+
+/**
+ * @title BucketLenderProxy
+ * @author dYdX
+ *
+ * TokenProxy for BucketLender contracts
+ */
+contract BucketLenderProxy
+{
+    using TokenInteract for address;
+    using AdvancedTokenInteract for address;
+
+    // ============ Constants ============
+
+    // Address of the WETH token
+    address public WETH;
+
+    // ============ Constructor ============
+
+    constructor(
+        address weth
+    )
+        public
+    {
+        WETH = weth;
+    }
+
+    // ============ Public Functions ============
+
+    /**
+     * Fallback function. Disallows ETH to be sent to this contract without data except when
+     * unwrapping WETH.
+     */
+    function ()
+        external
+        payable
+    {
+        require( // coverage-disable-line
```

Comment:
Something about how the gas is limited when sending Eth (to prevent reentrancy and stuff). The coverage will insert an event before and after this line, so that ends up hitting the gas limit in combination with `SLOAD`ing `WETH` from storage